pkg/vz/network_darwin.go

```diff
@@ -66,24 +66,9 @@ func forwardPackets(qemuConn *qemuPacketConn, vzConn *packetConn) {
 	defer qemuConn.Close()
 	defer vzConn.Close()
 
-	var wg sync.WaitGroup
-	wg.Add(2)
-
-	go func() {
-		defer wg.Done()
-		if _, err := io.Copy(qemuConn, vzConn); err != nil {
-			logrus.Errorf("Failed to forward packets from VZ to VMNET: %s", err)
-		}
-	}()
-
-	go func() {
-		defer wg.Done()
-		if _, err := io.Copy(vzConn, qemuConn); err != nil {
-			logrus.Errorf("Failed to forward packets from VMNET to VZ: %s", err)
-		}
-	}()
-
-	wg.Wait()
+	if err := bicopy.Bicopy(qemuConn, vzConn, "VMNET", "VZ"); err != nil {
+		logrus.WithError(err).Error("failed to forward packets")
+	}
```

Comment:
Previously we logged the error immediately when it happens. Now we my log no errors since after one copy fail, the second may be blocked forever. This will make it harder or impossible to debug issues.

I'm not sure that the current code is ensuring that if one copy fail the other one will also fail, and we don't have test for this error case. Before replacing the code lets understand if this is an issue and how we can fix it.

When we have 3 correct implementations we may consider unifying them.

Reply:
Sure. I moved all the logging back into bicopy.